plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -470,6 +475,33 @@ public void unregisterTable(ConnectorSession session, SchemaTableName tableName)
 
     @Override
     public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        if (isBigLake()) {
+            dropBigLakeTable(session, schemaTableName);
+        }
+        else {
+            purgeTable(session, schemaTableName);
+        }
+        invalidateTableCache(schemaTableName);
+        invalidateTableMappingCache(schemaTableName);
+    }
+
+    private void dropBigLakeTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        BaseTable table = loadTable(session, schemaTableName);
+        purgeTable(session, schemaTableName);
+        try {
+            // Explicitly remove data like TrinoGlueCatalog.dropTable since BigLake doesn't delete its data and metadata
+            dropTableData(table.io(), table.operations().current());
+        }
+        catch (RuntimeException e) {
+            // If the snapshot file is not found, an exception will be thrown by the dropTableData function.
```

Comment:
What's the exception here, FileNotFound or what, is it safe to declare that we are success here, no matter what kinda of the exception

Reply:
We want to suppress any exceptions since the table has been already dropped.